src/Versity.Users/Tests/Users.Tests/Application/RequestHandlers/Auth/ConfirmEmailRequestTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Application.Abstractions.Repositories;
+using Application.Exceptions;
+using Application.RequestHandlers.Auth.Commands.ConfirmEmail;
+using Bogus;
+using Domain.Models;
+using FluentAssertions;
+using Microsoft.AspNetCore.Identity;
+using Moq;
+
+namespace Users.Tests.Application.RequestHandlers.Auth;
+
+public class ConfirmEmailRequestTests
+{
+    private readonly Mock<IVersityUsersRepository> _versityUsersRepository;
+
+    public ConfirmEmailRequestTests()
+    {
+        _versityUsersRepository = new Mock<IVersityUsersRepository>();
+    }
+
+    [Fact]
+    public async Task RequestHandler_ShouldThrowException_WhenUserDoesNotExist()
+    {
+        // Arrange
+        _versityUsersRepository.Setup(
+            x => x.GetUserByIdAsync(It.IsAny<string>()))
+            .ReturnsAsync(null as VersityUser);
+        
+        var handler = new ConfirmEmailCommandHandler(_versityUsersRepository.Object);
```

Comment:
extract the creation of the handler to the constructor because it's setup logic and it'll reduce code duplication and check it through all files